tools/generator/AnDOGenerator.py

```diff
@@ -10,6 +15,117 @@
            ]
 
 
+class AnDOSesID:
+
+    def __init__(self, sesID=None, date=None, sesNumber=None, customSesField=None):
+        """
+        Representation of an AnDO session ID, as defined in `ando/rules/session_rules.json`
+
+        Parameters
+        ----------
+        sesID: (str)
+            complete session identifier, e.g. `20000101_001_test`
+        date: (str or datetime)
+            date subinfo of session ID, e.g. `20000101`
+        sesNumber: (str)
+            session number subinfo of session ID, e.g. `001`
+        customSesField: (str)
+            custom session field subinfo of session ID
+        """
+
+        # Inform user if insufficient or duplicate information is provided
+        if sesID is not None:
+            if (date is None) or (sesNumber is None) or (customSesField is None):
+                warnings.warn(f'Using sesID ({sesID}) in favour alternative parameters date, '
+                              f'sesNumber and customSesField')
+        elif (date is None) or (sesNumber is None):
+            raise ValueError('Incomplete information for generating a session ID.')
+
+
+        if sesID is None:
+            self.sesNumber = sesNumber
+
+            # Define default value for custom session field
+            if customSesField is None:
+                self.customSesField = ''
```

Comment:
I think I tested it with the checker yesterday and it didn't raise an error there. I am also not sure if it makes sense to force the user to have it since it artificially extends the folder names (and therefore potentially also the filenames later on). Let's have this discussion in a separate issue, since it is a general AnDO policy question. I openend #50 for this.

Reply:
I agree to discuss this in another issue!